### PR TITLE
custom input component

### PR DESCRIPTION
**Pull request by ayu-ayushisood**

### CODE REVIEW

File: habit/src/components/TextInput.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { TextInput } from 'react-native'
+import PropTypes from 'prop-types';
+
+export default function CustomTextInput(props){
+    return  <TextInput
```
Comment (karanasthana) [inline habit/src/components/TextInput.js:6]: Let's use prettier, to style our code properly

File: habit/src/components/screens/login/index.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
```
Comment (karanasthana) [inline habit/src/components/screens/login/index.js:1]: Move the file to ..src/containers/Login/LoginScreen ?

File: habit/src/components/TextInput.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { TextInput } from 'react-native'
```
Comment (karanasthana) [inline habit/src/components/TextInput.js:2]: Let's use the `native base` library's or `react-native-elements` library's text input component

File: habit/src/components/TextInput.js (hunk 1 above)
Comment (karanasthana) [reply habit/src/components/TextInput.js:6]: Also, share our prettier-rc files

File: habit/src/components/TextInput.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+import PropTypes from 'prop-types';
+
+export default function CustomTextInput(props){
+    return  <TextInput 
+                { ...props }
+                style={ props.style }
+            />
+}
+
+// CustomTextInput.propTypes = {
```
Comment (karanasthana) [inline habit/src/components/TextInput.js:12]: Let's remove this, if not needed

File: habit/src/components/screens/login/index.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+import { View, Text } from 'react-native'
+import CustomTextInput from '../../TextInput'
+
+export default function Login(props){
+    let input
+    return(
+        <View>
+            <Text>Hello login form</Text>
```
Comment (karanasthana) [inline habit/src/components/screens/login/index.js:9]: Let's use our custom text component

File: habit/src/components/screens/login/index.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { View, Text } from 'react-native'
+import CustomTextInput from '../../TextInput'
+
+export default function Login(props){
+    let input
```
Comment (karanasthana) [inline habit/src/components/screens/login/index.js:6]: Remove

File: habit/src/components/screens/login/index.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+    return(
+        <View>
+            <Text>Hello login form</Text>
+            <CustomTextInput 
+                style={{flex: 1}}
+                placeholder = 'Username'
+                ref={value => input = value}
```
Comment (karanasthana) [inline habit/src/components/screens/login/index.js:13]: What will we use the ref for?
Why not just store it into the state?

File: habit/src/components/screens/login/index.js [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+import { View, Text } from 'react-native'
+import CustomTextInput from '../../TextInput'
+
+export default function Login(props){
+    let input
+    return(
+        <View>
+            <Text>Hello login form</Text>
+            <CustomTextInput 
+                style={{flex: 1}}
```
Comment (karanasthana) [inline habit/src/components/screens/login/index.js:11]: Move this to a stylesheet

File: habit/src/components/screens/login/index.js [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+                placeholder = 'Username'
```
Comment (karanasthana) [inline habit/src/components/screens/login/index.js:12]: We can move the Username string to `constants` file

File: habit/src/components/screens/login/index.js [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+                ref={value => input = value}
+            />
+        </View>
+    )
```
Comment (karanasthana) [inline habit/src/components/screens/login/index.js:16]: Semi-colon

File: habit/src/components/screens/login/index.js (hunk 7 above)
Comment (karanasthana) [reply habit/src/components/screens/login/index.js:13]: Or will we use hooks?

File: habit/src/components/TextInput.js (hunk 4 above)
Comment (ayu-ayushisood) [reply habit/src/components/TextInput.js:12]: not now but the component is going to grow and we'll need it

File: habit/src/components/screens/login/index.js (hunk 2 above)
Comment (ayu-ayushisood) [reply habit/src/components/screens/login/index.js:1]: okay

File: habit/src/components/screens/login/index.js (hunk 8 above)
Comment (ayu-ayushisood) [reply habit/src/components/screens/login/index.js:11]: this is temporary

File: habit/src/components/screens/login/index.js (hunk 7 above)
Comment (ayu-ayushisood) [reply habit/src/components/screens/login/index.js:13]: ref is easy for these kind of small changes like input

File: habit/src/components/screens/login/index.js (hunk 10 above)
Comment (ayu-ayushisood) [reply habit/src/components/screens/login/index.js:16]: not important

File: habit/src/components/TextInput.js (hunk 1 above)
Comment (ayu-ayushisood) [reply habit/src/components/TextInput.js:6]: share, how?

File: habit/src/components/screens/login/index.js (hunk 5 above)
Comment (ayu-ayushisood) [reply habit/src/components/screens/login/index.js:9]: we are, this is temporary

File: habit/src/components/screens/login/index.js (hunk 6 above)
Comment (ayu-ayushisood) [reply habit/src/components/screens/login/index.js:6]: remove who?

File: habit/src/containers/login/index.js [hunk 11]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { View } from 'react-native'
+import CustomTextInput from '../../components/TextInput'
+
+export default function Login(props){
+    let [input, setInput] = React.useState('')
+    return(
+        <View>
+            <CustomTextInput 
+                style={{flex: 1}}
```
Comment (karanasthana) [inline habit/src/containers/login/index.js:10]: Move this to Stylesheet

File: habit/src/containers/login/index.js [hunk 12]
```diff
@@ -0,0 +1,17 @@
...
+                placeholder = 'Username'
```
Comment (karanasthana) [inline habit/src/containers/login/index.js:11]: Use a constant instead

Comment (karanasthana) [reply habit/src/containers/login/index.js:11]: in a file like `string_constants`

File: habit/src/components/TextInput.js (hunk 3 above)
Comment (ayu-ayushisood) [reply habit/src/components/TextInput.js]: done

File: habit/src/components/screens/login/index.js (hunk 8 above)
Comment (ayu-ayushisood) [reply habit/src/components/screens/login/index.js]: removed

File: habit/src/components/screens/login/index.js (hunk 9 above)
Comment (ayu-ayushisood) [reply habit/src/components/screens/login/index.js]: done

**Pull request merged**